python/tests/maps/test_sedonakepler_visualization.py

```diff
@@ -0,0 +1,28 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+from keplergl import KeplerGl
+from sedona.maps.SedonaKepler import SedonaKepler
+from tests.test_base import TestBase
+
+
+class TestVisualization(TestBase):
```

Comment:
Is it possible to add more tests for `SedonaKepler`, such as creating a map with an initial dataframe or adding a dataframe to an existing map?

Reply:
The test fails due to some known bugs of keplergl or geopandas (https://github.com/keplergl/kepler.gl/issues/1240), we need to work around this problem. We can check whether the data configuration with a specific dataId is added to the map config `kepler_map.config`, instead of checking the result of `kepler_map._repr_html()`.

Reply:
I do not think thats the case as the tests pass in my local machine, and here even the map_creation_with_df test fails which does not add the df separately.
I am investigating the root cause

Reply:
I cannot work with config as that would require me to create config objects for the map everytime I add a new df. As this bug only affects when adding the same dataset twice, I believe getting around this by adding a copy of the dataset is okay for these tests?

Reply:
I found that `_repr_html()` only works with pandas>=2.0.0. If we upgrade pandas in Pipfile, we have to drop tests for python 3.7.

If upgrading pandas is not an option, the best thing we can do is to check the keys of `kepler_map.data`, or simply don't validate the `kepler_map` since the constructor of `KeplerGl` has already validated the data. A successful map object construction indicates that everything is fine.

Reply:
I am able to get around this issue by exclusively using add_data of keplerGL instead of its df constructor. Please review this if the tests succeed @Kontinuation @jiayuasu